databroker/src/filter/filter_manager.rs

```diff
@@ -0,0 +1,256 @@
+/********************************************************************************
+* Copyright (c) 2025 Contributors to the Eclipse Foundation
+*
+* See the NOTICE file(s) distributed with this work for additional
+* information regarding copyright ownership.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Apache License 2.0 which is available at
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* SPDX-License-Identifier: Apache-2.0
+********************************************************************************/
+
+use std::collections::{BTreeSet, HashMap};
+use uuid::Uuid;
+
+use crate::types::{SignalId, TimeInterval};
+
+type SubscriptionUuid = Uuid;
+///
+/// FilterManager:
+/// Contains a small database with the signal_id, interval_ms and subscription_uuid
+///
+#[derive(Default)]
+pub struct FilterManager {
+    database: HashMap<SignalId, BTreeSet<(TimeInterval, SubscriptionUuid)>>,
+}
+
+impl FilterManager {
+    ///
+    /// Returns a HashMap containing each signal_id present in the map,
+    /// along with its corresponding lowest associated filter.
+    ///
+    fn get_lowest_filter_interval_per_signal(&self) -> HashMap<SignalId, TimeInterval> {
+        self.database
+            .clone()
+            .into_iter()
+            .filter_map(|(signal_id, set)| {
+                set.first()
+                    .map(|&(lowest_interval, _)| (signal_id, lowest_interval))
+            })
+            .collect()
+    }
+
+    ///
+    /// Add a new filter to each signal_id in the database and returns
+    /// the new update filter map containing ONLY signals_ids along with
+    /// their new lowest interval_ms.
+    ///
+    pub fn add_new_update_filter(
```

Comment:
align naming e.g. `add_new_update_filter` and `remove_filter_by_subscription_uuid` make it clear that this is the same filter we're talking about